cmd/print.go

```diff
@@ -164,15 +189,24 @@ func newPrintCommand() *cobra.Command {
 	var cmd = &cobra.Command{
 		Use:   "print",
 		Short: "Prints HTTPRoutes and Gateways generated from Ingress resources",
-		RunE:  pr.PrintGatewaysAndHTTPRoutes,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if pr.allResources && pr.inputFile == "" {
+				return errors.New("--all-resources flag can be set only when --input-file is set")
+			}
+			return nil
+		},
```

Comment:
The idea behind this is to properly convert a yaml file to the equivalent one with ingress resources replaced by Gateway API ones. I do not think that allowing this flag when reading from a cluster would be helpful in any case, as it'd basically be a dump of a cluster state into a unique giant print. Furthermore, we'd need a lot of custom logic to sanitize all the objects (e.g. status, timestamps, uid, etc.). I think we could leave this flag as it is for now, and if some use-case related to applying to cluster-based reading as well we can tackle it in a subsequent PR. What do you think?

Reply:
I agree, I also cant think of a good case where printing all from clusters will be useful. 
Also plus one that this can be addresses in a subsequent PR if ever requested.